emu_mps/mps_backend_impl.py

```diff
@@ -203,6 +214,18 @@ def init_initial_state(self, initial_state: State | None = None) -> None:
             )
 
         assert isinstance(initial_state, MPS)
+        if not torch.equal(
+            self.qubit_permutation, optimat.eye_permutation(self.qubit_count)
+        ):
+            # permute the initial state to match with permuted Hamiltonian
+            abstr_repr = initial_state._to_abstract_repr()
```

Comment:
this will fail if the initial state was not created via `MPS.from_state_amplitudes`, but for example as output from a `StateResult` observable. On a similar note, you don't reorder the state in a `StateResult` when reordering the results (for obvious reasons). I think we should add some logic that when optimatrix is used, `StateResult` is not allowed as an observable, and that if `initial_state._to_abstract_repr()` we add a more graceful error that state reordering is only supported when the input state is constructed from abstract representation.

Reply:
now `MPSconfig` checks observables and raises an error if the permutation if not supported